firmware/src/hal/sgx/src/trusted/secret_store.c

```diff
@@ -48,18 +49,64 @@
 
 #define MAX_BLOB_SIZE (1024 * 1024)
 
+#define SECRET_PAYLOAD_SIZE(secret_size) (secret_size - sizeof(secret_header_t))
+#define SEST_HASH_SIZE (32)
+
 // To avoid performing dynamic memory allocation whenever we need to read a
 // secret, we use this static buffer to temporarily store the sealed blobs we
 // read from the untrusted modules.
 static uint8_t G_secrets_buffer[MAX_BLOB_SIZE];
 
+// The sha256 context used for hash operations.
+static SHA256_CTX G_sha256_ctx;
+
+// The blob format of the sealed secrets we send to the seal API.
+// This includes the payload and a sha256 hash of the key for validation
+// purposes.
+typedef struct {
+    uint8_t key_hash[SEST_HASH_SIZE];
+} secret_header_t;
+
+typedef struct {
+    secret_header_t header;
+    uint8_t payload[0];
+} secret_t;
+
 // The file format of the sealed secrets. This is the format we use to read and
 // write sealed the data to disk.
 typedef struct {
     size_t blob_size;
     uint8_t* blob;
 } sealed_secret_t;
 
+/**
+ * @brief Sets the secret header for the given key.
+ *
+ * @param key The key to hash and set the header for.
+ * @param header The destination for the secret header.
+ */
+static void set_secret_header(const char* key, secret_header_t* header) {
+    sha256_init(&G_sha256_ctx);
+    sha256_update(&G_sha256_ctx, (const uint8_t*)key, strlen(key));
+    sha256_final(&G_sha256_ctx, header->key_hash);
+}
+
+/**
+ * @brief Checks if the given secret header is valid for the given key.
+ *
+ * @param key The key to validate the header against.
+ * @param header The header to validate.
+ *
+ * @returns true if the header is valid, false otherwise.
+ */
+static bool check_secret_header(const char* key,
```

Comment:
Please rename to `is_secret_header_valid` to improve readability.

Reply:
sounds good